File: samples/client/petstore/java/src/main/java/org/openapijsonschematools/client/schemas/validation/NumberSchemaValidator.java
package org.openapijsonschematools.client.schemas.validation;

import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;

public interface NumberSchemaValidator<T> {
    Number validate(Number arg, SchemaConfiguration configuration) throws ValidationException;
    T validateAndBox(Number arg, SchemaConfiguration configuration) throws ValidationException;
}

File: samples/client/3_1_0_unit_test/kotlin/src/main/kotlin/org/openapijsonschematools/client/schemas/validation/NumberSchemaValidator.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

interface NumberSchemaValidator<T> {
    @Throws(ValidationException::class)
    fun validate(arg: Number, configuration: SchemaConfiguration?): Number

    @Throws(ValidationException::class)
    fun validateAndBox(arg: Number, configuration: SchemaConfiguration?): T
}